.github/workflows/ci.yml

```diff
@@ -12,26 +12,36 @@ jobs:
   run-test:
     timeout-minutes: 15
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: ["22.x"]
+    name: Build with Node ${{ matrix.node-version }}
     steps:
-      - name: Checkout Source
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
-      - name: Set up Node
-        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
```

Comment:
@arunanshub You should ping the `pnpm/action-setup` to a commit sha. This is because a tag such as `v4` can be overwritten. If `pnpm/action-setup` is compromised, attacker will overwrite `v4` with a malicious code. Pinning the version to a commit sha makes 3rd party dependencies immutable.